src/block_cache/block_io.c

```diff
@@ -313,11 +388,19 @@ __wt_blkcache_read_multi(WT_SESSION_IMPL *session, WT_ITEM **buf, size_t *buf_co
      */
     dsk = results[0].data;
     if (F_ISSET(dsk, WT_PAGE_ENCRYPTED)) {
-        WT_ASSERT(session, session == NULL);
+        WT_CLEAR(etmp);
+        WT_RET(__read_decrypt(session, &results[0], &etmp, addr, addr_size, false));
+        WT_ITEM_MOVE(results[0], etmp);
+    } else if (btree->kencryptor != NULL) {
+        WT_ERR(__blkcache_read_corrupt(
+          session, WT_ERROR, addr, addr_size, "unencrypted block for which encryption configured"));
     }
 
+    dsk = results[0].data;
     if (F_ISSET(dsk, WT_PAGE_COMPRESSED)) {
-        WT_ASSERT(session, session == NULL);
+        WT_CLEAR(ctmp);
+        WT_RET(__read_decompress(session, dsk, &ctmp, addr, addr_size));
+        WT_ITEM_SET(results[0], ctmp);
```

Comment:
Here it is still wrong?

Reply:
Can I resolve this? This section got rewritten.